jujugui/static/gui/src/app/components/machine-view/machine-view.js

```diff
@@ -106,8 +106,18 @@ YUI.add('machine-view', function() {
       @param {Object} unit The unit that was dropped.
       @param {String} machine The machine id that the unit dropped onto.
     */
-    _dropUnit: function(unit, machine) {
-      this.props.placeUnit(unit, machine);
+    _dropUnit: function(unit, machine, newType) {
+      if (machine) {
+        this.props.placeUnit(unit, machine);
+      } else {
+        var state = {};
+        if (newType === 'machine') {
+          state.showAddMachine = {unit: unit};
```

Comment:
`showAddMachine` and `showAddContainer` are Boolean when defined but here you're passing an object. It's typically considered bad practice to change types. Can this be reworked to not change the type?